### ISSUE TITLE
Allow `WorklowDefinition`s and `Workflow`s to hold metadata

### ISSUE THREAD

**Issue by masinger** [lisegmbh/fluxflow#87]
Similar to StepDefinitions and Jobs, metadata should also be available for `WorkflowDefinition`s and `Workflow`s.

They should also be automatically constructed from annotations present at the relevant locations.

Relates to #86 and #50.

**Comment by bobmazy** [comment]
@masinger 
While reviewing the code, I noticed that both the Job and JobDefinition interfaces include a metadata field:
- Job.metaData ([Job.kt](https://github.com/lisegmbh/fluxflow/blob/1a8428320143ad989fd2352cf36d8c8abcd82e66/library/core/api/src/main/kotlin/de/lise/fluxflow/api/job/Job.kt#L20))
- JobDefinition.metadata ([JobDefinition.kt](https://github.com/lisegmbh/fluxflow/blob/1a8428320143ad989fd2352cf36d8c8abcd82e66/library/core/api/src/main/kotlin/de/lise/fluxflow/api/job/JobDefinition.kt#L19))

Both fields store the same metadata, and in [ReflectedJobDefinition](https://github.com/lisegmbh/fluxflow/blob/1a8428320143ad989fd2352cf36d8c8abcd82e66/library/core/stereotyped/src/main/kotlin/de/lise/fluxflow/stereotyped/job/ReflectedJobDefinition.kt#L38) the metadata from the definition is passed to the ReflectedJob.

Is this duplication intentional? Or would it be beneficial to refactor the design to avoid redundant metadata storage?

**Comment by masinger** [comment]
Yes, the duplication is by design. The idea behind this is to enable developer to dynamically add custom metadata at runtime.

All definition metadata is currently derived from annotations and therefore kind of static. Whenever a new job/step/workflow is started, the definition's metadata is copied to the the activated workflow object. A developer can then add custom metadata to these objects. See https://docs.fluxflow.cloud/en/stable/steps/#modifying-step-metadata-at-runtime
